hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/clustering/HoodieFlinkClusteringJob.java

```diff
@@ -358,6 +358,9 @@ private void cluster() throws Exception {
           .setParallelism(1);
 
       env.execute("flink_hudi_clustering_" + clusteringInstant.getTimestamp());
+      if (conf.getBoolean(FlinkOptions.CLEAN_ASYNC_ENABLED) && conf.getBoolean(FlinkOptions.CLEAN_OFFLINE_ENABLE)){
+        writeClient.clean();
+      }
```

Comment:
CompactionCommitSink#open should already handle the cleaning.

Reply:
CleanFunction#open only do clean when OptionsResolver.isInsertOverwrite(conf) is true.

  @Override
  public void open(Configuration parameters) throws Exception {
    super.open(parameters);
    if (conf.getBoolean(FlinkOptions.CLEAN_ASYNC_ENABLED)) {
      this.writeClient = FlinkWriteClients.createWriteClient(conf, getRuntimeContext());
      this.executor = NonThrownExecutor.builder(LOG).waitForTasksFinish(true).build();

      if (OptionsResolver.isInsertOverwrite(conf)) {
        String instantTime = HoodieActiveTimeline.createNewInstantTime();
        LOG.info(String.format("exec sync clean with instant time %s...", instantTime));
        executor.execute(() -> writeClient.clean(instantTime), "wait for sync cleaning finish");
      }
    }
  }